### PR TITLE
[MIG] crm_lead_firstname: Migration to 13.0

### PR DESCRIPTION
**Pull request by ntsirintanis**

### CODE REVIEW

**Comment by rousseldenis** [comment]
@ntsirintanis Could you try to rebase ?

**Comment by NL66278** [comment]
@ntsirintanis Travis fails because of the precommit hook. Please black your code, preferably by using the pre-commit hook from the maintainer quality tools, before submitting again. Runbot fails due to running for more then an hour. This might be caused by other modules running very long to test. Unfortunately there is no detailed log available.

**Comment by NL66278** [comment]
@ntsirintanis These are the fatal (ERROR) runbot errors:
```
[91mERROR: pylint-odoo 0.0.1.dev459 has requirement lxml>=4.2.3, but you'll have lxml 3.7.1 which is incompatible.
[0m[91mERROR: coveralls 1.8.2 has requirement coverage<5.0,>=3.6, but you'll have coverage 5.0.3 which is incompatible.
```

**Comment by hbrunn** [comment]
I think https://travis-ci.org/OCA/crm/jobs/654812753#L849 is.

You need https://github.com/OCA/crm/blob/12.0/oca_dependencies.txt

**Comment by NL66278** [comment]
@ntsirintanis and the oca-dependencies.txt file is missing. You can include this from the 12.0 branch as well. This contains the reference to the partner-contact repository.

**Comment by ntsirintanis** [comment]
OK, I'll see into correcting these failing unit tests

**Comment by hbrunn** [comment]
seems to me you'll have to migrate them to the new form stuff you were struggling with when trying to fix `partner_firstname`'s tests. If you choose your tests wisely, you should be able to circumvent the bug triggered by `partner_firstname`, as the underlying issue is not yet fixed to my knowledge

File: crm_lead_firstname/tests/test_crm_lead.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        self.assertEqual(self.lead.contact_lastname, partner.lastname)
+
+    def test_create_contact_empty(self):
+        """No problems creating a contact without names."""
+        self.lead.write({"contact_name": False, "contact_lastname": False})
+        self.lead.handle_partner_assignation()
+
+    def test_onchange_partner(self):
+        """When changing partner, fields get correctly updated."""
+        with self.env.do_in_onchange():
```
Comment (simahawk) [inline crm_lead_firstname/tests/test_crm_lead.py]: tests are failing [here](https://travis-ci.org/github/OCA/crm/jobs/657385009#L845).
You should use the form class https://github.com/odoo/odoo/blob/13.0/odoo/tests/common.py#L1336
(search for examples in the code, is plenty).

**Comment by simahawk** [comment]
hmm the commits sounds wrong: crm_lead_product is already in v13.
You should:
1. hard reset on 13.0
2. cherry-pick your migration commits
3. fix bugs

**Comment by ntsirintanis** [comment]
@simahawk thanks a lot for your comments and review! I had mixed up some commits, but this is resolved now. Also fixed unit tests. Let's see if this gets green now.

**Comment by ntsirintanis** [comment]
@hbrunn @NL66278 @rousseldenis @simahawk LGTM please.

**Review by lfreeke — APPROVED** [review]
Functional test :+1:

**Review by hbrunn — APPROVED** [review]
you can add yourself to the contributors

File: crm_lead_firstname/tests/test_crm_lead.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import Form, SavepointCase
+
+
+class FirstNameCase(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(FirstNameCase, cls).setUpClass()
+        cls.lead_model = cls.env["crm.lead"]
```
Comment (simahawk) [inline crm_lead_firstname/tests/test_crm_lead.py:12]:
```suggestion
        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
        cls.lead_model = cls.env["crm.lead"]
```

File: crm_lead_firstname/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
```
Comment (simahawk) [inline crm_lead_firstname/__manifest__.py]: replace `©`with `Copyright` all around pls

File: crm_lead_firstname/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Firstname and Lastname in Leads",
+    "summary": "Specify split names for contacts in leads",
+    "version": "13.0.1.0.0",
+    "category": "Customer Relationship Management",
+    "website": "http://www.github.com/OCA/crm",
```
Comment (simahawk) [inline crm_lead_firstname/__manifest__.py]:
```suggestion
    "website": "http://github.com/OCA/crm",
```

File: crm_lead_firstname/readme/INSTALL.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Since leads are expected to create partners only when needed and after
```
Comment (simahawk) [inline crm_lead_firstname/readme/INSTALL.rst]: I'd move this paragraph to DESCRIPTION.rst

**Review by simahawk — CHANGES_REQUESTED** [review]
LG, just a bunch of small changes please :)

**Comment by ntsirintanis** [comment]
@simahawk please take another look.

File: crm_lead_firstname/readme/INSTALL.rst (hunk 5 above)
Comment (simahawk) [reply crm_lead_firstname/readme/INSTALL.rst]: @ntsirintanis you forgot to remove it here ;)

Comment (ntsirintanis) [reply crm_lead_firstname/readme/INSTALL.rst]: Damn, that's what happens when you forget to ctrl + s after editing! Thanks

**Review by simahawk — APPROVED** [review]

**Pull request merged**